HopsanGUI/HcomHandler.cpp

```diff
@@ -7820,10 +7820,13 @@ QString HcomHandler::runScriptCommands(QStringList &lines, bool *pAbort)
                 HCOMERR("If requires one condition expression");
                 return "";
             }
-            QString condition = args.front();
-            QStringList ifCode;
+            QList<QString> conditions;
+            conditions.append(args.front());
+            QList<QStringList> ifCodes;
+            ifCodes.append(QStringList());
             QStringList elseCode;
             bool inElse=false;
+            int depth=1;
```

Comment:
Is see that only depth one is processed, but just want to see if i understand correctly.
Nested if statements (depth >=2) would be interpreted runScriptCommands(ifCodes[i] is called.
But every time depth will be at most one.

Reply:
Yes, exactly. Nested loops will be treated iteratively in the next evaluation. This makes it possible to have an infinite depth of nested statements.